apps/ledger-live-desktop/src/renderer/analytics/segment.ts

```diff
@@ -55,38 +55,30 @@ export function setAnalyticsFeatureFlagMethod(method: typeof analyticsFeatureFla
   analyticsFeatureFlagMethod = method;
 }
 
+let isBatch1Enabled: boolean,
+  isBatch2Enabled: boolean,
+  isBatch3Enabled: boolean,
+  ptxEarnEnabled: boolean,
+  stakingProvidersEnabled: number | string;
+
 const getFeatureFlagProperties = () => {
   if (!analyticsFeatureFlagMethod) return {};
-  (async () => {
-    const { id } = await user();
-    const analytics = getAnalytics();
-    const ptxEarnFeatureFlag = analyticsFeatureFlagMethod("ptxEarn");
-    const fetchAdditionalCoins = analyticsFeatureFlagMethod("fetchAdditionalCoins");
-    const stakingProviders = analyticsFeatureFlagMethod("ethStakingProviders");
+  const ptxEarnFeatureFlag = analyticsFeatureFlagMethod("ptxEarn");
+  const fetchAdditionalCoins = analyticsFeatureFlagMethod("fetchAdditionalCoins");
+  const stakingProviders = analyticsFeatureFlagMethod("ethStakingProviders");
 
-    const isBatch1Enabled =
-      !!fetchAdditionalCoins?.enabled && fetchAdditionalCoins?.params?.batch === 1;
-    const isBatch2Enabled =
-      !!fetchAdditionalCoins?.enabled && fetchAdditionalCoins?.params?.batch === 2;
-    const isBatch3Enabled =
-      !!fetchAdditionalCoins?.enabled && fetchAdditionalCoins?.params?.batch === 3;
-    const stakingProvidersEnabled =
-      stakingProviders?.enabled && stakingProviders?.params?.listProvider.length;
+  isBatch1Enabled = !!fetchAdditionalCoins?.enabled && fetchAdditionalCoins?.params?.batch === 1;
+  isBatch2Enabled = !!fetchAdditionalCoins?.enabled && fetchAdditionalCoins?.params?.batch === 2;
+  isBatch3Enabled = !!fetchAdditionalCoins?.enabled && fetchAdditionalCoins?.params?.batch === 3;
+  ptxEarnEnabled = !!ptxEarnFeatureFlag?.enabled;
+  stakingProvidersEnabled =
+    !!stakingProviders?.enabled &&
+    stakingProviders?.params &&
+    stakingProviders?.params?.listProvider?.length > 0
+      ? stakingProviders?.params?.listProvider.length
+      : "flag not loaded";
```

Comment:
love it great idea, much preferable to global vars

Reply:
in the process of updating this pr i realised that the `getFeatureFlagProperties` fn no longer adds traits to the user, so it wasn't actually doing anything with them except calling `updateIdentify`. So I've keep the `callOnce` of updateIdentify after the flags are definitely loaded, since that seems harmless, but added. the traits back to the extra properties. The `getPtxAttributes` fn may be useful again if some of them do need to be added to the user traits.